### ISSUE TITLE
AppImage: Mux server not starting mux server

### ISSUE THREAD

**Issue by linsomniac** [wez/wezterm#2039]
### What Operating System(s) are you seeing this problem on?

Linux Wayland, macOS

### WezTerm version

20220408-101518-b908e2dd

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

I'm using a MacOS client side and Ubuntu Linux 22.04 laptop as the server side (called "nlaptop" below).

I've had this successfully work, without manually starting the mux server, on my initial install of wezterm.  But later it started failing as described below.  The difference between the two runs that I am aware of is:

- First run, worked, but I had just installed wezterm and no wezterm processes were running on the Linux side.
- Second run, failed, but I had switched over to using wezterm on the Linux side and had ~10 wezterms running on the Linux console before I tried to start the mux connection from the Mac.

When I try to "wezterm connect nlaptop" I get the following message:

    05:14:12.316  INFO   wezterm_gui::termwindow > OpenGL initialized! Apple M1 Max 4.1 Metal - 76.3 is_context_loss_possible=false wezterm version: 20220408-101518-b908e2dd
    05:14:12.347  ERROR  wezterm_client::client  > going to run /home/sean/bin/wezterm cli proxy
    05:14:12.991  ERROR  wezterm_client::client  > ssh stderr: 05:14:13.050  ERROR  wezterm > connecting to /run/user/1000/wezterm/sock: Connection refused (os error 111); terminating
    05:14:13.102  ERROR  wezterm_client::client  > /home/sean/bin/wezterm cli proxy failed
    05:14:13.104  ERROR  wezterm_client::client  > Error while decoding response pdu: decoding a PDU: reading PDU length: EOF while reading leb128 encoded value
    05:14:13.104  ERROR  wezterm_gui             > Please install the same version of wezterm on both the client and server! The server reported error 'Error while decoding response pdu: decoding a PDU: reading PDU length: EOF while reading leb128 encoded value' while being asked for its version.  This likely means that the server is older than the client.
    ; terminating
    05:14:13.104  ERROR  mux::connui             > while running ConnectionUI loop: recv_timeout: channel is empty and disconnected

On the Linux side, this produces a log file in `/run/user/1000/wezterm/wezterm-log-1075738.txt` with the contents:

    05:23:15.448  ERROR  wezterm > connecting to /run/user/1000/wezterm/sock: Connection refused (os error 111); terminating

If I ssh into the Linux machine and run `wezterm cli --prefer-mux spawn` and then on the Mac I run `wezterm connect nlaptop`, that works as expected.

### To Reproduce

I'm guessing:

- Using AppImage on Linux side.
- Shut down all wezterm processes.
- Start some local wezterms on the mux server box.
- On remote machine do "wezterm connect $OTHER_BOX".

### Configuration

return {
  ssh_domains = {
    {
      name = "nlaptop",
      remote_address = "nlaptop",
      local_echo_threshold_ms = 400,
      multiplexing = "WezTerm",
      remote_wezterm_path = "/home/sean/bin/wezterm",
      username = "sean",
    }
  },
}

### Expected Behavior

wezterm mux starts up as part of the "wezterm connect" from the Mac side.

### Logs

05:23:15.448  ERROR  wezterm > connecting to /run/user/1000/wezterm/sock: Connection refused (os error 111); terminating

**Comment by philipGaudreau** [comment]
I'm having the same issue. It seems to be related to credentials.

**Comment by wez** [comment]
@linsomniac: sorry for taking a while to get back to you on this.

I suspect something weird with AppImage and resolving things from the path. wezterm wants to spawn `wezterm-mux-server` to start the server, but I suspect that, since your have the appimage installed in your home dir, some combination of:
* wezterm cannot find the path to wezterm in the context of the ssh session (I know you specify the path to it in your config)
* wezterm cannot find the path to `wezterm-mux-server`, which is embedded in the appimage and only visible from inside it.

There's some somewhat complicated environment munging that happens to support running the appimage:
https://github.com/wez/wezterm/blob/main/env-bootstrap/src/lib.rs#L15
and I suspect that it may be getting confused in a nested invocation situation like bootstrapping the mux server.

I think you may need to install a non-appimage build for this to work 100% headache-free.

Or another possibility: try symlinking `wezterm-mux-server` to the `wezterm` appimage and see if that magically helps.

**Comment by philipGaudreau** [comment]
For me, it had to do with my `default_prog`, which was not available on the remote machine. I saw the error when I tried to launch `wezterm-mux-server` on the server, without the `--daemonize` option.

**Comment by linsomniac** [comment]
Thanks for following up @wez, I grabbed the latest stable (June 22-ish) and installed that on both my Mac and Linux boxes, on Linux I removed the appimage and installed the .deb.  I also changed the paths in my wezterm config.  After that, I was able to "wezterm connect tlaptop", and have it come up.  So far it's looking good.  I will let you know if I have any further problems.

Also, that makes wezterm start much faster, with the appimage it takes 2-4 seconds, with the deb it is instantaneous.  I knew that was a tradeoff of the appimage, but I don't start new terminals very often.
